ZFitter/src/addBranch_class.cc

```diff
@@ -515,7 +515,61 @@ TTree* addBranch_class::AddBranch_smearerCat(TChain* originalChain, TString tree
 	return newtree;
 }
 
+TTree* addBranch_class::AddBranch_scaleEle(TChain* originalChain, TString treename, bool isMC){
+  //To do:
+  //Add flexibility in .dat file name
+  //Add flexibility in branch names
+  //how to change doScale BEFORE creating escaler (otherwise default is false and if doScale is false correction file is not parsed)-->can't we just change doScale=true by default
+  EnergyScaleCorrection_class eScaler("Calibration/ZFitter/ScalesSmearings/Winter_2016_reReco_v1_ele",0,true,false);
+  originalChain->SetBranchStatus("*",0);
+  originalChain->SetBranchStatus("runNumber",1);
+  originalChain->SetBranchStatus("etaSCEle",1);
+  originalChain->SetBranchStatus("R9Ele",1);
+  originalChain->SetBranchStatus("energy_ECAL_ele",1);
```

Comment:
you need energyBranchName to be a string given to addBranch_class in the constructor or using a method setting a private member. Don't need to active all the energy branches